support-lib/jni/djinni_support.cpp

```diff
@@ -22,6 +22,7 @@
 #include <cstring>
 
 static_assert(sizeof(jlong) >= sizeof(void*), "must be able to fit a void* into a jlong");
+static_assert(sizeof(wchar_t) == 2 || sizeof(wchar_t) == 4, "wchar_t must be represented by UTF-16 or UTF-32 encoding");
```

Comment:
C++ standard has no any guarantees about wchar_t representation. But almost all modern compilers uses utf-16 or utf-32 to represent wide characters and many serious projects adopt such assumptions (for example, Python`s unicode object: https://github.com/python/cpython/blob/2.7/Objects/unicodeobject.c).